Code/EnginePlugins/JoltPlugin/System/JoltQueries.cpp

```diff
@@ -327,14 +327,56 @@ void ezJoltWorldModule::QueryShapesInSphere(ezPhysicsOverlapResultArray& out_res
     return;
 
   const JPH::SphereShape shape(fSphereRadius);
+  const JPH::Mat44 trans = JPH::Mat44::sTranslation(ezJoltConversionUtils::ToVec3(vPosition));
+
+  QueryShapes(out_results, shape, trans, params);
+}
+
+void ezJoltWorldModule::QueryShapesInBox(ezPhysicsOverlapResultArray& out_results, const ezVec3& vBoxExtents, const ezTransform& transform, const ezPhysicsQueryParameters& params) const
+{
+  out_results.m_Results.Clear();
+
+  if (vBoxExtents.x <= 0.0f || vBoxExtents.y <= 0.0f || vBoxExtents.z <= 0.0f)
+    return;
+
+  const JPH::BoxShape shape(ezJoltConversionUtils::ToVec3(vBoxExtents * 0.5f));
+  const JPH::Mat44 trans = JPH::Mat44::sRotationTranslation(ezJoltConversionUtils::ToQuat(transform.m_qRotation),
+                                                            ezJoltConversionUtils::ToVec3(transform.m_vPosition));
+
+  QueryShapes(out_results, shape, trans, params);
+}
+
+void ezJoltWorldModule::QueryShapesInCapsule(ezPhysicsOverlapResultArray& out_results, float fCapsuleRadius, float fCapsuleHeight, const ezTransform& transform, const ezPhysicsQueryParameters& params) const
+{
+  out_results.m_Results.Clear();
+
+  if (fCapsuleRadius <= 0.0f)
+    return;
+
+  const JPH::CapsuleShape shape(fCapsuleHeight * 0.5f, fCapsuleRadius);
```

Comment:
you can multiply transform.m_vScale.z into the capsule height, and ezMath::Max(m_vScale.x, m_vScale.y) into the radius